### PR TITLE
Builds with Intel PIN 3.4

### PR DESCRIPTION
**Pull request by BeauJoh**
But had to dirtily hack a bug with branch prediction monitoring in the ppm code.
This edition of PIN segfaults when parsing `xbegin` and `xend` instructions on branch monitoring.

### CODE REVIEW

**Comment by boegel** [comment]
@amirjamez Please review?

File: mica_ilp.cpp [hunk 1]
```diff
@@ -138,9 +138,9 @@ void init_ilp_one(){
 			cerr << "ERROR! Interval size is not a multiple of ILP buffer size. (" << interval_size << " vs " << ILP_BUFFER_SIZE << ")" << endl;
 			exit(-1);
 		}
-		stringstream filename;
-		filename << "ilp-win" << win_size << "_phases_int";
-		output_file_ilp_one.open(mkfilename(filename.str().c_str()), ios::out|ios::trunc);
+		char filename[100];
+        sprintf(filename, "ilp-win%d_phases_int", win_size);
```
Comment (boegel) [inline mica_ilp.cpp]: what's up with the funky indentation here?

File: mica_ilp.cpp [hunk 2]
```diff
@@ -207,9 +207,10 @@ VOID ilp_instr_intervals_one(){
 
 	if(interval_ins_count_for_hpc_alignment == interval_size){
 
-		stringstream filename;
-		filename << "ilp-win" << win_size << "_phases_int";
-		output_file_ilp_one.open(mkfilename(filename.str().c_str()), ios::out|ios::app);
+        char filename[100];
+        sprintf(filename, "ilp-win%d_phases_int", win_size);
```
Comment (boegel) [inline mica_ilp.cpp]: same here, weird indentation?

File: mica_ilp.cpp [hunk 3]
```diff
@@ -408,18 +409,19 @@ VOID writeMem_ilp_one(ADDRINT effAddr, ADDRINT size){
...
+    char filename[100];
 
 	fini_ilp_buffering_one();
 
 	if(interval_size == -1){
-		filename << "ilp-win" << win_size << "_full_int";
-		output_file_ilp_one.open(mkfilename(filename.str().c_str()), ios::out|ios::trunc);
+        sprintf(filename, "ilp-win%d_full_int", win_size);
+
+        output_file_ilp_one.open(mkfilename(filename), ios::out|ios::trunc);
```
Comment (boegel) [inline mica_ilp.cpp]: more weird indentation

File: mica_ilp.cpp [hunk 4]
```diff
@@ -408,18 +409,19 @@ VOID writeMem_ilp_one(ADDRINT effAddr, ADDRINT size){
...
 		output_file_ilp_one << total_ins_count;
 	}
 	else{
-		filename << "ilp-win" << win_size << "_phases_int";
-		output_file_ilp_one.open(mkfilename(filename.str().c_str()), ios::out|ios::app);
+        sprintf(filename, "ilp-win%d_phases_int", win_size);
+        output_file_ilp_one.open(mkfilename(filename), ios::out|ios::app);
```
Comment (boegel) [inline mica_ilp.cpp]: even more weird indentation

File: mica_init.cpp [hunk 5]
```diff
@@ -7,6 +7,10 @@
  * information.
  */
 
+//#include <cstdint>
+//#include <sys/mount.h>
+//#include <limits.h>
```
Comment (boegel) [inline mica_init.cpp]: commented out, so not needed?

File: mica_ppm.cpp [hunk 6]
```diff
@@ -532,8 +532,9 @@ void register_condBr(ADDRINT ins_addr){
 	indices_condBr[numStatCondBranchInst++] = ins_addr;
 }
 
+static int _count  = 0;
 VOID instrument_ppm_cond_br(INS ins){
-	UINT32 index = index_condBr(INS_Address(ins));
+    UINT32 index = index_condBr(INS_Address(ins));
```
Comment (boegel) [inline mica_ppm.cpp]: more weird indentation?

File: mica_ppm.cpp [hunk 7]
```diff
@@ -544,8 +545,21 @@ VOID instrument_ppm_cond_br(INS ins){
...
+    if (strncmp(str, substr, strlen(substr)) == 0){
+        printf("as of pin 3.4 -- I don't think we can parse xbegin so skipping...\n");
+        return;
+    }
+    substr = "xend";
+    if (strncmp(str, substr, strlen(substr)) == 0){
+        printf("as of pin 3.4 -- I don't think we can parse xend so skipping...\n");
+        return;
+    }
+    INS_InsertCall(ins, IPOINT_BEFORE, (AFUNPTR)condBr,IARG_UINT32, index, IARG_BRANCH_TAKEN, IARG_END);
```
Comment (boegel) [inline mica_ppm.cpp]: same here, more weird/inconsistent indentation...

File: mica_ilp.cpp (hunk 1 above)
Comment (amirjamez) [reply mica_ilp.cpp]: That's right. Sort of a Pythonish indentation.

**Pull request merged**